Allow to send messages to a list of KNX group addresses

open-homeautomation opened the pull request:
## Description:

Allows to send a message not only to a single group address, but also to multiple group addresses.
The message_listener has been converted to a service as automations can only call services, but not send messages
## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

Commit:
Allow to send messages to a list of KNX group addresses

Commit:
Allow to send messages to a list of KNX group addresses

Commit:
Formatting

Commit:
Formatting

Commit:
Converted message interface to a service

Commit:
Converted message interface to a service

pvizeli commented on homeassistant/components/knx.py:
```diff
@@ -120,13 +126,14 @@ def handle_knx_send(event):
 
     # Listen for when knx_frame_send event is fired
-    hass.bus.listen(EVENT_KNX_FRAME_SEND, handle_knx_send)
+    hass.services.register(DOMAIN, 'group_write', handle_group_write)
```
Please add a vol schema and remove checks inside your service function.

open-homeautomation replied:
You mean a vol schema for the parameters of the service function?

open-homeautomation replied:
BTW: Is it possible with vol to have an argument that is either a string a a list of strings?

pvizeli replied:
yes. Can you make a example, and I can help with this schema

amelchio committed:
LIFX: make broadcast address configurable (#8453)

andrey-git committed:
Backend support for themes (#8419)

* Backend support for themes

* Fix test

* Add theme_updated event

* Shorten name

* Add tests

Danielhiversen committed:
upgrade broadlink (#8462)

Danielhiversen committed:
upgrade rfxtrx lib (#8463)

abmantis committed:
Plex: Add exception handler when connection fails (#8179)

* add exception handler when connection fails

* plex: improve exception handling

* remove uneeded exception handler

danielperna84 committed:
HomeMatic dependency upgrade + IP Wall Thermostat support (#8465)

* Upgrade dependency + added IP Wall Thermostat

andrey-git committed:
Add kiosk-mode panel (#8457)

balloob committed:
Update frontend

balloob committed:
Fix iFrame panel test

MartinHjelmare committed:
Make deps directory persistent over upgrades (#7801)

* Use pip install --user if venv not active

* Set PYTHONUSERBASE to deps directory, when installing with --user
  option.
* Reset --prefix option to workaround incompatability when installing
  with --user option. This requires pip version 8.0.0 or greater.
* Require pip version 8.0.3.
* Do not delete deps directory on home assistant upgrade.
* Fix local lib mount and check package exist.

* Update and add tests

* Fix upgrade from before version 0.46

* Extract function to get user site

* Add function(s) to package util to get user site.
* Use async subprocess for one of the functions to get user site.
* Add function to package util to check if virtual environment is
  active.
* Add and update tests.

* Update version for last removal of deps dir

* Address comments

* Rewrite package util tests with pytest

* Rewrite all existing unittest class based tests for package util as
  test functions, and capitalize pytest fixtures.
* Add test for installing with target inside venv.

amelchio committed:
LIFX: support for multizone (#8399)

* Make aiolifx modules easily available

* Use aiolifx features_map for deciding bulb features

Also move the feature detection out of Light so it is available even
during the initial detection.

* Move each LIFX light type to a separate class

* Simplify AwaitAioLIFX

This has become possible with recent aiolifx that calls the callback even
when a message is lost.

Now the wrapper can be used also before a Light is added though the register
callback then has to become a coroutine.

* Refactor send_color

* Add support for multizone

This lets lifx_set_state work on individual zones.

Also update to aiolifx_effects 0.1.1 that restores the state for individual
zones.

d0ugal committed:
Add RGB support to switch.flux (#8417)

dersger committed:
Fix media_position for cast component (#8452)

* Make it available during state paused.

* Don't adjust for media_position_updated_at. I.e. do as vlc, sonos etc
  so that returned position is the position at the time of
  media_position_updated_at, not now.

gollo committed:
Remove km from visibility, add visibility_distance (#8454)

* Remove km from visibility, add visibility_distance

* Fix line length

* Fix trailing space and line break indentation

* Indentation

* More whitespace

hoopty committed:
Radarr sensor fix for issue #8250 (#8456)

* Radarr sensor fix for issue #8250

* Radarr sensor fix for issue #8250

amelchio committed:
Accept transition for light.toggle (#8466)

andrey-git committed:
Make themes API work even when themes are not defined. (#8473)

MartinHjelmare committed:
Fix check for running inside venv (#8481)

* Import and use the function from pip instead of defining it
  ourselves.
* Fix tests.

balloob committed:
Version bump to 0.50.0dev0

Commit:
Uses vol to check service parameters
Added descirption of service parameters

Commit:
Merge branch 'knxmsg3' of https://github.com/open-homeautomation/home-assistant into knxmsg3
